subprojects/docs/src/docs/userguide/dependencyManagementJavaProjects.adoc

```diff
@@ -0,0 +1,109 @@
+apiElements::
+The dependencies which are part of this project's externally consumable API as well as the classes which are defined in this project which should be consumable by other projects. This configuration is an example of exposing artifacts for consumption.
+
+Various plugins add further standard configurations. You can also define your own custom configurations in your build via api:org.gradle.api.Project#configurations[]. See <<defining_and_using_configurations>> for the details of defining and customizing dependency configurations.
+
+[[sec:repositories_java_tutorial]]
+=== Declaring repositories
+
+How does Gradle know where to find the files for external dependencies? Gradle looks for them in a _repository_. A repository is a collection of modules, organized by `group`, `name` and `version`. Gradle understands different <<repository_types,repository types>>, such as Maven and Ivy, and supports various ways of accessing the repository via HTTP or other protocols.
+
+By default, Gradle does not define any repositories. You need to define at least one with the help of api:org.gradle.api.Project#repositories[] before you can use module dependencies. One option is use the Maven Central repository:
+
+++++
+<sample id="defineMavenCentral" dir="userguide/artifacts/defineRepository" title="Usage of Maven central repository">
+    <sourcefile file="build.gradle" snippet="maven-central"/>
+</sample>
+++++
+
+You can also have repositories on the local file system. This works for both Maven and Ivy repositories.
+
+++++
+<sample id="defineRemoteIvyRepo" dir="userguide/artifacts/defineRepository" title="Usage of a local Ivy directory">
+    <sourcefile file="build.gradle" snippet="local-ivy-repo"/>
+</sample>
+++++
+
+A project can have multiple repositories. Gradle will look for a dependency in each repository in the order they are specified, stopping at the first repository that contains the requested module.
+
+To find out more about defining repositories, have a look at <<declaring_repositories>>.
+
+[[sec:publishing_artifacts_java_tutorial]]
+=== Publishing artifacts
+
+Dependency configurations are also used to publish files. Gradle calls these files _publication artifacts_, or usually just _artifacts_. As a user you will need to tell Gradle where to publish the artifacts. You do this by declaring repositories for the `uploadArchives` task. Here's an example of publishing to a Maven repository:
+
+++++
+<sample id="publishMavenRepository" dir="userguide/artifacts/maven" title="Publishing to a Maven repository">
+    <sourcefile file="build.gradle" snippet="upload-file"/>
+</sample>
+++++
+
+Now, when you run `gradle uploadArchives`, Gradle will build the JAR file, generate a `.pom` file and upload the artifacts.
+
+To find out more about publication, have a look at <<artifact_management>>.
```

Comment:
"To learn more about publishing artifacts, have a look at ..."

Reply:
done